Cabal/Distribution/Simple/GHC/Internal.hs

```diff
@@ -531,13 +531,15 @@ simpleGhcEnvironmentFile packageDBs pkgids =
 --
 -- The 'Platform' and GHC 'Version' are needed as part of the file name.
 --
+-- Returns the name of the file written.
 writeGhcEnvironmentFile :: FilePath  -- ^ directory in which to put it
                         -> Platform  -- ^ the GHC target platform
                         -> Version   -- ^ the GHC version
                         -> [GhcEnvironmentFileEntry] -- ^ the content
-                        -> NoCallStackIO ()
-writeGhcEnvironmentFile directory platform ghcversion =
-    writeFileAtomic envfile . BS.pack . renderGhcEnvironmentFile
+                        -> NoCallStackIO FilePath
+writeGhcEnvironmentFile directory platform ghcversion entries = do
+    writeFileAtomic envfile . BS.pack . renderGhcEnvironmentFile $ entries
+    return envfile
```

Comment:
If I wanted to be picky, I'd create a pure function to compute the envfile name, and then have `writeGhcEnvironmentFile` take the final filename to write it out to. You don't have to do this.